src/core/pretty.ml

```diff
@@ -49,7 +49,9 @@ let rec pp_p_term : p_term pp = fun oc t ->
   let out fmt = Format.fprintf oc fmt in
   let rec pp p _ t =
     let pp_env _ ar =
-      if Array.length ar > 0 then out "[%a]" (Array.pp (pp PFunc) ", ") ar
+      match ar with
+      | None -> ()
+      | Some ar -> out "[%a]" (Array.pp (pp PFunc) ", ") ar
```

Comment:
Note that, now, when translating Dk files to LP files, pattern variables will always be followed by square brackets even if it has no argument. To change this, one needs to know whether we went through an abstraction or not.

Reply:
Fixed in 4e9dc0e71f36adf8b0be816a9c292d3e65f15434